granulate_utils/linux/ns.py

```diff
@@ -59,19 +59,23 @@ def resolve_proc_root_links(proc_root: str, ns_path: str) -> str:
     parts = Path(ns_path).parts
 
     path = proc_root
+    seen = set()
     for part in parts[1:]:  # skip the / (or multiple /// as .parts gives them)
         next_path = os.path.join(path, part)
-        if os.path.islink(next_path):
+        while os.path.islink(next_path):
+            if next_path in seen:
+                raise OSError("Symlink loop detected")
```

Comment:
I don't like this use of `OSError` (a standard exception with expected fields such as `errno`) for non standard errors. I think that `Exception` suffices here if we don't plan to have a specialized exception to help catching it.

Reply:
I can raise `OSError(errno.ELOOP)` which is raised from `os.path.realpath` if you want. I can also raise `RuntimeError("Symlink loop from %r" % e.filename)` which is what's raised from `pathlib.Path.resolve` as a wrapper for the `OSError`. The second one is probably better.

Reply:
We can go with the second one